src/main/java/ca/on/oicr/gsi/dimsum/service/filtering/DateFilterKey.java

```diff
@@ -0,0 +1,54 @@
+package ca.on.oicr.gsi.dimsum.service.filtering;
+
+import java.time.LocalDate;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import ca.on.oicr.gsi.dimsum.data.RequisitionQc;
+import ca.on.oicr.gsi.dimsum.data.Sample;
+
+
+public enum DateFilterKey {
+  // @formatter:off
+  AFTER_DATE(string -> {
+    return sample -> sample.getQcDate() != null &&!sample.getQcDate().isBefore(LocalDate.parse(string)); 
```

Comment:
For run-libraries (if sample.run is not null), data review date should be used instead of QC date. Same in `BEFORE_DATE`